Source/ACE.DatLoader/FileTypes/CharGen.cs

```diff
@@ -0,0 +1,232 @@
+﻿using ACE.DatLoader.Entity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ACE.DatLoader.FileTypes
+{
+    public class CharGen
+    {
+        private static CharGen instance;
+
+        public int Did { get; set; }
+        public List<List<Loc>> StarterAreas { get; set; } = new List<List<Loc>>();
+        public Dictionary<int, HeritageGroupCG> HeritageGroups { get; set; } = new Dictionary<int, HeritageGroupCG>();
+
+        public static CharGen ReadFromDat(DatReader datReader)
+        {
+            if (instance == null) // We'll store the CG data into the instance the first time it's loaded. No need to read it multiple times.
+            {
+                CharGen cg = new CharGen();
+
+                cg.Did = datReader.ReadInt32();
+                datReader.Offset = 8;
+
+                /* STARTER AREA */

```

Comment:
Multi-line comment style is not really necessary for single line comments, but I do like how this shows separate logic sections.

You can use /// for a lot of metadata in comments that will be visible within visual studio.